chain/stmgr/stmgr.go

```diff
@@ -51,9 +51,10 @@ type versionSpec struct {
 }
 
 type migration struct {
-	upgrade       MigrationFunc
-	preMigrations []PreMigration
-	cache         *nv16.MemMigrationCache
+	upgrade            MigrationFunc
+	preMigrations      []PreMigration
+	cache              *nv16.MemMigrationCache
+	migratedStateroots map[cid.Cid]cid.Cid
```

Comment:
Map here is for reorgs that might make us run the same migration on different stateroots, we want to make sure we grab the correct migrated root for each input.